x-pack/plugins/observability/server/services/slo/slo_repository.ts

```diff
@@ -85,23 +84,17 @@ export class KibanaSavedObjectsSLORepository implements SLORepository {
     await this.soClient.delete(SO_SLO_TYPE, response.saved_objects[0].id);
   }
 
-  async findAllByIds(ids: string[]): Promise<SLO[]> {
+  async findAllByIds(ids: string[], namespace?: string): Promise<SLO[]> {
     if (ids.length === 0) return [];
 
-    try {
-      const response = await this.soClient.find<StoredSLO>({
-        type: SO_SLO_TYPE,
-        page: 1,
-        perPage: ids.length,
-        filter: `slo.attributes.id:(${ids.join(' or ')})`,
-      });
-      return response.saved_objects.map((slo) => toSLO(slo.attributes));
-    } catch (err) {
-      if (SavedObjectsErrorHelpers.isNotFoundError(err)) {
-        throw new SLONotFound(`SLOs [${ids.join(',')}] not found`);
-      }
-      throw err;
-    }
+    const response = await this.soClient.find<StoredSLO>({
+      type: SO_SLO_TYPE,
+      page: 1,
+      perPage: ids.length,
+      filter: `slo.attributes.id:(${ids.join(' or ')})`,
+      namespaces: namespace ? [namespace] : undefined,
```

Comment:
Is that available when the soclient is bound to a user's request?

Reply:
in that case you don't need it, since client is already bounded to space.